packages/jsii-pacmak/test/expected.jsii-calc/java/src/main/java/software/amazon/jsii/tests/calculator/Calculator.java

```diff
@@ -18,14 +18,71 @@ protected Calculator(final software.amazon.jsii.JsiiObject.InitializationMode in
         super(initializationMode);
     }
 
+    /**
+     * A builder facility for {@link Calculator}.
+     */
+    public static final class Builder {
+        private software.amazon.jsii.tests.calculator.CalculatorProps.Builder props;
+
+        /**
+         * Creates a new {@link Builder} for {@link Calculator}.
+         *
+         *
+         * @returns a new {@link Builder} instance
+         */
+        public static Builder create() {
+            return new Builder();
+        }
+
+        private Builder() {
+        }
+
+        /**
+         * Defines the value of props.initialValue
+         *
+         * @param initialValue 
+         *
+         * @returns {@code this}
+         */
+        public Builder initialValue(final java.lang.Number initialValue) {
+            if (this.props == null) {
+                this.props = software.amazon.jsii.tests.calculator.CalculatorProps.builder();
+            }
+            this.props.initialValue(initialValue);
+            return this;
+        }
+
+        /**
+         * Defines the value of props.maximumValue
+         *
+         * @param maximumValue 
+         *
+         * @returns {@code this}
+         */
+        public Builder maximumValue(final java.lang.Number maximumValue) {
+            if (this.props == null) {
+                this.props = software.amazon.jsii.tests.calculator.CalculatorProps.builder();
```

Comment:
If there are lots of nested properties it will look bad to duplicate this code over and over, maybe think about creating a private method to do this lazy initialization.